sdk/search/azure-search-documents/azure/search/documents/_service/_models.py

```diff
@@ -87,3 +91,57 @@ def __init__(self, **kwargs):
         self.pattern = kwargs.get("pattern", r"\W+")
         self.flags = kwargs.get("flags", None)
         self.group = kwargs.get("group", -1)
+
+
+class DataSource(_DataSource):
```

Comment:
I think I can prolly remove this.

Another option is to create a model which accepts credentials and pass it to the datasource credentials  - but not sure if it brings more value other than not exposing an additional generated model.

Reply:
Agree I don't really see value in it as-is. For reference I was thinking more along the lines of `SuggestQuery` etc, that are not themselves Models, but do some helper/convenience work to create the actual model objects more simply for users.

Reply:
@rakshith91 did you still want to remove this before merge?

Reply:
removed for now